docs/core/metrics.md

```diff
@@ -224,14 +239,15 @@ You can add high-cardinality data as part of your Metrics log with `add_metadata
     --8<-- "examples/metrics/src/add_metadata_output.json"
     ```
 
-### Single metric with a different dimension
+### Single metric
 
-CloudWatch EMF uses the same dimensions across all your metrics. Use `single_metric` if you have a metric that should have different dimensions.
+CloudWatch EMF uses the same dimensions and timestamp across all your metrics. Use `single_metric` if you have a metric that should have different dimensions or timestamp.
```

Comment:
This sounds weird to me. If I just use the metrics powertools module without specifying any timestamp, and my function takes 60 seconds, will all the metrics generated have the same millisecond?

Reply:
If I understand [the spec](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html) correctly, they will have the timestamp of the time the EMF blob was created & flushed out of the buffer. Unless you’re using the `SingleMetric` - which is flushed immediately - when calling `add_metric` we put these metric in a buffer. 

At that stage the metric doesn’t have any timestamp but only the values for name, unit, etc. then, later, when they are flushed either manually by calling the appropriate method or automatically by the decorator logic, the EMF blob is emitted:

```json
{
  "_aws": {
    "Timestamp": 1559748430481,
    "CloudWatchMetrics": [ ... ]
  }
}
```

Dimensions behave similarly in the sense that they’re not specific to a single metric but apply to the entire EMF blob:

```json
{
  "_aws": {
    "Timestamp": 1559748430481,
    "CloudWatchMetrics": [
      {
        "Dimensions": [ [ "functionVersion" ] ],
        "Namespace": "serverlessAirline ",
        "Metrics": [ … ] <- actual metrics from add_metric go here
      }
    ]
  }
}
```

Reply:
Great point. I guess the only thing to think about (outside of the scope) is when we are batching a big number of metrics, we are generating a slightly different timestamp for each batch. But should all be inside the same second (thinking of high resolution metrics) so it should be fine.